src_c/surface.c

```diff
@@ -2029,13 +2041,19 @@ surf_blits(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
         if (doreturn) {
             retrect = NULL;
             retrect = pgRect_New(&dest_rect);
-            if (PyList_Append(ret, retrect) != 0) {
-                bliterrornum = BLITS_ERR_PY_EXCEPTION_RAISED;
-                goto bliterror;
+            if (issequence) {
+                PyList_SET_ITEM(ret, curriter, retrect);
+            }
```

Comment:
I think this would work if you put the `curriter++` inside this block here. `PyList_SET_ITEM(ret, curriter++, retrect)`

This keeps the curriter incrementing closer to its area of responsibility, and doesn't spend time doing that on iterators of undefined length.

Reply:
yep agree